rskj-core/src/main/java/co/rsk/Start.java

```diff
@@ -33,11 +33,17 @@
     public static void main(String[] args) {
         setUpThread(Thread.currentThread());
 
+        RskContext ctx = null;
         try {
-            RskContext ctx = new RskContext(args);
+            ctx = new RskContext(args);
             runNode(Runtime.getRuntime(), new PreflightChecksUtils(ctx), ctx);
         } catch (Exception e) {
             logger.error("The RSK node main thread failed, closing program", e);
+
+            if (ctx != null) {
+                ctx.close();
+            }
+
             System.exit(1);
```

Comment:
Maybe we can put all of this inside a finally block?

Reply:
we should close it only if an exception has been thrown. If no exception, we the context should stay non-closed until a process finishes.

Reply:
There's a runtime hook in `runNode` making sure the context is closed on normal shutdown. Actually, I think in case of most exceptions, it will be called twice, but should be ok because it is idempotent.